src/com/projectkorra/projectkorra/firebending/bluefire/BlueFirePassive.java

```diff
@@ -0,0 +1,80 @@
+package com.projectkorra.projectkorra.firebending.bluefire;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import com.projectkorra.projectkorra.GeneralMethods;
+import com.projectkorra.projectkorra.ability.BlueFireAbility;
+import com.projectkorra.projectkorra.ability.PassiveAbility;
+import com.projectkorra.projectkorra.util.TempBlock;
+
+public class BlueFirePassive extends BlueFireAbility implements PassiveAbility {
+
+	public BlueFirePassive(Player player) {
+		super(player);
+		// TODO Auto-generated constructor stub
+
+	}
+
+	@Override
+	public void progress() {
+		// TODO Auto-generated method stub
+		if(bPlayer.canBendPassive(this)) {
+			for(Block b : GeneralMethods.getBlocksAroundPoint(player.getLocation(), 5)) {
+				if (b.getType() == Material.TORCH ) { 
```

Comment:
Switch statement?

Reply:
Might be beneficial, but originally I was planning on checking not just the material but the block state for Campfires in order to prevent item loss, if we want to add that in the future it'll be hard w/ a switch statement.